src/Bundle.ts

```diff
@@ -12,68 +12,349 @@ import transformBundle from './utils/transformBundle';
 import collapseSourcemaps from './utils/collapseSourcemaps';
 import callIfFunction from './utils/callIfFunction';
 import error from './utils/error';
-import {
-	dirname,
-	isRelative,
-	isAbsolute,
-	normalize,
-	relative,
-	resolve
-} from './utils/path';
+import { normalize, resolve } from './utils/path';
 import { OutputOptions } from './rollup/index';
 import { RawSourceMap } from 'source-map';
 import Graph from './Graph';
 import ExternalModule from './ExternalModule';
-import ExternalVariable from './ast/variables/ExternalVariable';
 import ExportDefaultVariable from './ast/variables/ExportDefaultVariable';
+import Variable from './ast/variables/Variable';
+import NamespaceVariable from './ast/variables/NamespaceVariable';
+import ExternalVariable from './ast/variables/ExternalVariable';
+import { makeLegal } from './utils/identifierHelpers';
 import { DynamicImportMechanism } from './ast/nodes/Import';
 
+export type BundleDependencies = {
+	id: string;
+	name: string;
+	isBundle: boolean;
+	reexports?: ReexportSpecifier[];
+	imports?: ImportSpecifier[];
+}[];
+
+export type BundleExports = {
+	local: string;
+	exported: string;
+}[];
+
+export interface ReexportSpecifier {
+	reexported: string;
+	imported: string;
+};
+export interface ImportSpecifier {
+	local: string;
+	imported: string;
+};
+
 export default class Bundle {
+	id: string;
+	name: string;
 	graph: Graph;
 	orderedModules: Module[];
+
+	// this represents the bundle module wrappings
+	// which form the output dependency graph
+
+	// map from variable exported by this bundle to its safe exported name
+	exportedVariables: Map<Variable, string>;
+	imports: {
+		module: Bundle | ExternalModule;
+		variables: {
+			// the name of the export this import corresponds to
+			name: string;
+			module: Module | ExternalModule;
+			variable: Variable;
+		}[];
+	}[];
+	exports: {
+		[safeName: string]: {
+			// module can be in or out of bundle
+			// if module is out of the bundle then it is a reexport
+			module: Module | ExternalModule;
+			// exported name from source module
+			name: string;
+			variable: Variable;
+		}
+	}
+	dependencies: (ExternalModule | Bundle)[];
 	externalModules: ExternalModule[];
+	// an entry module bundle is a bundle that exactly exports the exports of
+	// an input entry point module
 	entryModule: Module;
+	entryModuleFacade: boolean;
 
-	constructor (graph: Graph, orderedModules: Module[]) {
+	constructor (graph: Graph, id: string, orderedModules: Module[]) {
+		this.setId(id);
 		this.graph = graph;
 		this.orderedModules = orderedModules;
+
+		this.exportedVariables = new Map();
+		this.imports = [];
+		this.exports = {};
 		this.externalModules = undefined;
+
+		this.dependencies = undefined;
 		this.entryModule = undefined;
+		this.entryModuleFacade = false;
+		orderedModules.forEach(module => {
```

Comment:
Could make sense to put this into a separate method, e.g. `analyseModules`? It is quite possible more code will be added here in the future.

Reply:
The analysis is all in the separate "collect" / "generate" functions. This is only for the entryModule and entryModuleFacade properties.